### PR TITLE
runfix: Correctly check if user has mls device before performing mls operations

### PR DESCRIPTION
**Pull request by atomrc**
## Description

see https://github.com/wireapp/wire-web-packages/pull/6185/files

## Checklist

- [x] PR has been self reviewed by the author;
- [x] Hard-to-understand areas of the code have been commented;
- [x] If it is a core feature, unit tests have been added;

### CODE REVIEW

File: src/script/main/app.ts [hunk 1]
```diff
@@ -443,7 +443,7 @@ export class App {
       // We load all the users the self user is connected with
       await userRepository.loadUsers(selfUser, connections, conversations, teamMembers);
 
-      if (supportsMLS()) {
+      if (this.core.hasMLSDevice) {
```
Comment (atomrc) [inline src/script/main/app.ts:446]: if we have a MLS device it means that:
- the env supportsMLS (previous check)
- the team has MLS enabled
- the user has a valid mls device

(it a more exhaustive check than the previous `supportsMLS` check)

File: src/script/conversation/ConversationVerificationStateHandler/MLS/MLSStateHandler.ts [hunk 2]
```diff
@@ -58,12 +58,12 @@ export class MLSConversationVerificationStateHandler {
   ) {
     this.logger = getLogger('MLSConversationVerificationStateHandler');
     // We need to check if the core service is available and if the e2eIdentity is available
-    if (!this.core.service?.mls || !this.core.service?.e2eIdentity) {
+    if (!this.core.hasMLSDevice || !this.core.service?.e2eIdentity) {
```
Comment (V-Gira) [inline src/script/conversation/ConversationVerificationStateHandler/MLS/MLSStateHandler.ts]: is this correct? If so, it doesn't match the code comment anymore right?

File: src/script/main/app.ts [hunk 3]
```diff
@@ -472,7 +472,7 @@ export class App {
 
       await conversationRepository.init1To1Conversations(connections, conversations);
 
-      if (supportsMLS()) {
+      if (this.core.hasMLSDevice) {
```
Comment (V-Gira) [inline src/script/main/app.ts:475]: this one is the fix to automation correct?
Seems like the change in `MLSConversationVerificationStateHandler` isn't necessarry? (that's the one breaking unit tests)

File: src/script/conversation/ConversationVerificationStateHandler/MLS/MLSStateHandler.ts (hunk 2 above)
Comment (atomrc) [reply src/script/conversation/ConversationVerificationStateHandler/MLS/MLSStateHandler.ts]: indeed the comment needs updating, good catch :) 
The fix is needed though (we should never check if mls is enabled by checking that the service is there, anymore)

Comment (V-Gira) [reply src/script/conversation/ConversationVerificationStateHandler/MLS/MLSStateHandler.ts]: gotcha, I guess we're just not spying on that `hasMLSDevice` in the unit test then

File: src/script/main/app.ts (hunk 3 above)
Comment (atomrc) [reply src/script/main/app.ts:475]: exactly, I'm working on the fix right now :)

**Review by V-Gira — APPROVED** [review]

**Pull request merged**